hail/src/main/scala/is/hail/expr/types/physical/PBinary.scala

```diff
@@ -71,56 +67,42 @@ class PBinary(override val required: Boolean) extends PType {
       }
     }
   }
-
-  override def byteSize: Long = 8
-
-  override def containsPointers: Boolean = true
 }
 
 object PBinary {
-  def apply(required: Boolean = false): PBinary = if (required) PBinaryRequired else PBinaryOptional
+  def apply(required: Boolean = false): PBinary = PCanonicalBinary.apply(required)
```

Comment:
This is a different way of doing what we did with PString. For consistency, we should either choose the way we did this with PString, or the way we did this here. Advantage of the present solution (in this PR) is that PCanonicalBinaryRequired and PCanonicalBinaryOptional exist (whereas they didn't for PCanonicalString). If the goal is really to have a single instance of PBinary floating around, then the present solution (in this PR) is the better approach.